CentralComputing/TCPManager.cpp

```diff
@@ -55,14 +55,30 @@ int TCPManager::read_command(Network_Command * buffer) {
   return bytes_read;
 }
 
-int TCPManager::write_data() {
+int TCPManager::write_data(vector<int64_t>& times) {
+  vector<int32_t> vals;
+  if(Utils::microseconds() - times[0] > 1000000){  //  This is the first time threshold
+    vals.push_back(POD_STATE);
+    vals.push_back(POSITION); 
+    vals.push_back(VELOCITY);
+    vals.push_back(ACCELERATION);
+    times[0] = Utils::microseconds();
+  }
+  if(Utils::microseconds() - times[1] > 3000000){  //  This is the second time threshold 
+    vals.push_back(TEMPERATURE);
+    times[1] = Utils::microseconds();
+  }
+  if(Utils::microseconds() - times[2] > 6000000){  //  This is the third time threshold 
```

Comment:
Ideally, there is some way to turn these `if`s into something that you can put into a for loop, since all of the conditions in the `if` can be parameterized to an index. 

I'm not sure exactly how you would get the `vals.push_back` to be parameratized by index, and it might not be worth the effort. 

For now, this is a very convienent and easy way of doing things.